Clean up comments and misspellings

kgaillot opened the pull request:
This small request updates some obsolete comments and misspellings.

kgaillot commented:
I ran pengine/regression.sh and CTS with 100 random tests. CTS has an issue with broadcast-fencing nodes that appears unrelated to these changes, and I am investigating that separately, but the other 97 tests succeeded.

beekhof commented on cib/remote.c:
```diff
@@ -324,7 +324,9 @@ cib_remote_listen(gpointer data)
     crm_client_init();
     new_client = calloc(1, sizeof(crm_client_t));
+    CRM_ASSERT(new_client != NULL);
     new_client->remote = calloc(1, sizeof(crm_remote_t));
```
What have you gained here?
Accessing a NULL pointer will give you the same effect.

kgaillot replied:
I think a log message is more helpful to a sysadmin than a core dump if a problem does occur, though I agree that the chances of the issue are tiny.

beekhof replied:
You'd be surprised. Most cannot tell the difference.

kgaillot replied:
After a bit of research, I found that in the C standard, dereferencing a NULL pointer results in "undefined behavior".

From http://en.wikipedia.org/wiki/Null_pointer: "Dereferencing the NULL pointer typically results in ... a segmentation fault ... There are, however, certainly circumstances where this is not the case. ... It is also possible for the compiler to optimize away the `NULL` pointer dereference, avoiding a segmentation fault but causing other undesired behavior. ... an attempt to dereference a null pointer usually (but not always) causes a run-time error or immediate program crash."

From http://blog.llvm.org/2011/05/what-every-c-programmer-should-know.html: "contrary to popular belief, dereferencing a null pointer in C is undefined. It is not defined to trap ... In C-based languages, NULL being undefined enables a large number of simple scalar optimizations that are exposed as a result of macro expansion and inlining."

beekhof replied:
There is a big difference between being undefined in the C standard and undefined in the various implementations.
At the point `gcc` or `clvm` change their behaviour, I will agree to sprinkling the code with asserts.

Until then, we have other fish to fry :)

kgaillot commented:
This was previously the memory-check request. I popped off the memory-check revision, updated the two minor revisions and rebased it on current master.
